### PR TITLE
tests: add spread tests for registries

### LINKED ISSUE #5054: Add a spread test for registries
### What needs to get done

Add a spread test for editing and listing registries, similar to the spread test for validation sets

### Why it needs to get done

For end-to-end testing

Depends on #5020

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the [guidelines for contributing](https://github.com/canonical/snapcraft/blob/main/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox run -m lint`?
- [x] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

Add a spread test for `edit-registries` and `list-registries`.

Fixes #5054
(CRAFT-3419)

### CODE REVIEW

File: tests/spread/store/registries/task.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  snap install yq
+  # registries only available in edge
+  snap refresh snapd --edge
+
+execute: |
+  # ensure snapcraft is logged in and can access the store
+  snapcraft whoami
+
+  # snapcraft will use a fake file editor
```
Comment (tigarmo) [inline tests/spread/store/registries/task.yaml:33]: clever

**Review by tigarmo — APPROVED** [review]

File: tests/spread/store/registries/task.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  fi
+
+  # setup snap gpg dir
+  mkdir -p "$HOME/.snap/gnupg"
+  chmod 700 "$HOME/.snap/gnupg"
+
+  # import a registered key
+  echo "$SNAPCRAFT_ASSERTION_KEY" | base64 --decode > store-key.txt
+  gpg --homedir "$HOME/.snap/gnupg" --import store-key.txt
+  rm -f store-key.txt
```
Comment (dariuszd21) [inline tests/spread/store/registries/task.yaml:23]: It's from the top of my head, but it should work 
```suggestion
  gpg --homedir "$HOME/.snap/gnupg" --import <(echo "$SNAPCRAFT_ASSERTION_KEY" | base64 --decode)
```

**Review by dariuszd21 — APPROVED** [review]
Looks good!

File: tests/spread/store/registries/task.yaml (hunk 2 above)
Comment (mr-cal) [reply tests/spread/store/registries/task.yaml:23]: I tried a one-liner before and gpg was being weird about stdin.  I'll give this a try

Comment (mr-cal) [reply tests/spread/store/registries/task.yaml:23]: I probably tried a pipe instead of a redirect

**Comment by mr-cal** [comment]
Failures unrelated - network related errors for ros2 deps

**Pull request merged**